File: 백준/Silver/10845. 큐/큐.py
import sys

cnt=int(input())
queue=[]

for i in range(cnt):
    orders=list(sys.stdin.readline().split())
    order=orders[0]
    if order=='push':
        queue.append(int(orders[1]))

    if order=='pop':
        if len(queue)==0:
            print(-1)
            continue
        print(queue.pop(0))

    if order=='size':
        print(len(queue))

    if order=='empty':
        if len(queue)==0: print(1)
        else: print(0)

    if order=='front':
        if len(queue)==0:
            print(-1)
            continue
        print(queue[0])
        

    if order=='back':
        if len(queue)==0:
            print(-1)
            continue
        print(queue[len(queue)-1])

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int orderCnt = Integer.parseInt(br.readLine());
		
		Deque<String> queue = new LinkedList();
		for (int i = 0; i < orderCnt; i++) {
			String[] order = br.readLine().split(" ");
			switch(order[0]) {
				case "push":
					queue.add(order[1]);
					break;
				case "pop":
					System.out.println(queue.isEmpty()?-1:queue.remove());
					break;
				case "size":
					System.out.println(queue.size());
					break;
				case "empty":
					System.out.println(queue.isEmpty()?1:0);
					break;
				case "front":
					System.out.println(queue.isEmpty()?-1:queue.peek());
					break;
				case "back":
					System.out.println(queue.isEmpty()?-1:queue.peekLast());
					break;
			}
		}
	}
}